common/djangoapps/third_party_auth/saml.py

```diff
@@ -93,3 +95,158 @@ def wrapped_method(*args, **kwargs):
     def _config(self):
         from .models import SAMLConfiguration
         return SAMLConfiguration.current(Site.objects.get_current(get_current_request()))
+
+
+class SapSuccessFactorsIdentityProvider(SAMLIdentityProvider):
```

Comment:
@bradenmacdonald, actually that adds complexity. There is a `get_user_details` method on the backend, but it's not isomorphic with the `get_user_details` method on the identity provider; you can see the base code here:

https://github.com/python-social-auth/social-core/blob/master/social_core/backends/saml.py#L269

Notably, none of the details we actually need to get off the configuration are present on the backend object - they're stored on the retrieved IdP. If we were to put the methods on the backend, we'd be passing back either the IdP, or at least its conf, as a big ball of state between all those methods. In comparison, by subclassing the IdP, the config variables are natively part of the details saved to the instance, and we can access them from anywhere.